packages/extractor/src/extractor/mod.js

```diff
@@ -181,9 +187,9 @@ class Extractor {
 	 * @returns {ts.Program}
 	 */
 	#create_program() {
-		this.#cache.root_names.add(this.compiler.filepath);
```

Comment:
`root_names` keeps growing with each extract, leading to unintended reuse of previous sources.